src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,165 @@
+package parser;
+
+import command.Command;
+import exception.DukeException;
+import tasklist.TaskList;
+import ui.Ui;
+import task.Task;
+import task.Todo;
+import task.Deadline;
+import task.Event;
+
+import java.text.ParseException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+import java.util.NoSuchElementException;
+import java.util.Scanner;
+
+/**
+ * Parser class used to process Strings into variables that the Command object can execute.
+ */
+public class Parser {
+
+    /** Command portion of the input. */
+    private String commandString;
+    /** Description portion of the input. */
+    private String desc;
+    /** Date when the task starts. */
+    private LocalDate startDate;
+    /** Time when the task starts. */
+    private LocalTime startTime;
+    /** Time when the task ends. */
+    private LocalTime endTime;
+    /** Index for deletion or marking tasks as done. */
+    private int index;
+    /** Keyword for finding the task in a list. */
+    private String keyword;
+
+
+    /**
+     * Parses user input into a Command object containing its required components such as description, date and time
+     *      according to the command given.
+     * @param userInput Input provided by the user.
+     * @throws DukeException Exception is thrown when there is an invalid format provided. For example,
+     *      missing date / time or incorrect date / time format.
+     */
+    public static Command parse(String userInput) throws DukeException {
+        Scanner sc = new Scanner(userInput);
+        Command newCommand = new Command();
+        String commandString = sc.next();
+        newCommand.addCommandString(commandString);
+
+        switch (commandString) {
+        case "find":
+            newCommand.addKeyword(sc.next());
```

Comment:
Will there be a possible "NoSuchElementException" error? If command follows with a blank.

Reply:
Yo thanks for raising this up! Did this awhile back and forgot about it haha. Fixed it already thanks! 😀